keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/LaunchConfiguration.kt

```diff
@@ -27,34 +27,8 @@ data class LaunchConfiguration(
   val instanceMonitoring: Boolean = false,
   val ramdiskId: String? = null
 ) {
-  fun toLaunchConfigurationSpec(account: String, application: String, omitDefaults: Boolean = false) =
-    ClusterSpec.LaunchConfigurationSpec(
-      instanceType = instanceType,
-      ebsOptimized = if (omitDefaults && !ebsOptimized) {
-        null
-      } else {
-        ebsOptimized
-      },
-      iamRole = if (omitDefaults && iamRole == defaultIamRoleForApp(application)) {
-        null
-      } else {
-        iamRole
-      },
-      keyPair = if (omitDefaults && keyPair == defaultKeyPairForAccount(account)) {
-        null
-      } else {
-        keyPair
-      },
-      instanceMonitoring = if (omitDefaults && !instanceMonitoring) {
-        null
-      } else {
-        instanceMonitoring
-      },
-      ramdiskId = ramdiskId
-    )
-
   companion object {
-    fun defaultKeyPairForAccount(account: String) = "nf-$account-keypair-a"
+    fun defaultKeyPairForAccount(account: String) = "nf-keypair-$account-{{region}}"
```

Comment:
My understanding is that the "latest and greatest" convention used by newer accounts, hence the change here. This will apply during resolution only.

Reply:
We shouldn't have to care about this in keel. Clouddriver will return the default keypair for the account. 
We need to handle expanding the `{{region}}` templating in the right place in keel.

Reply:
I'm confused as to why this is still present given `clouddriverCache.defaultKeyPairForAccount()`.

Reply:
@emjburns @asher These are used in this block of `ClusterSpec.resolveLaunchConfiguration()`:
https://github.com/spinnaker/keel/blob/a456b51eb6eb9fb35796a1c86601b87de22dc40b/keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt#L52-L57

I added them because I noticed that `iamRole` and `keyPair` were defaulting to `null` during resolution, and I thought we actually wanted to enforce those conventions when creating resources... If that's not the case, I'll just remove both functions and those references.

Reply:
I'm considering this resolved based on @robfletcher's feedback below: https://github.com/spinnaker/keel/pull/593#discussion_r346449171